include/seqan3/io/sequence_file/format_genbank.hpp

```diff
@@ -0,0 +1,252 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the seqan3::sequence_file_format_genbank class.
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <iterator>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <range/v3/view/chunk.hpp>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/io/detail/misc.hpp>
+#include <seqan3/io/sequence_file/input_options.hpp>
+#include <seqan3/io/sequence_file/output_options.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+#include <seqan3/range/detail/misc.hpp>
+#include <seqan3/range/view/char_to.hpp>
+#include <seqan3/range/view/to_char.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/range/view/take_line.hpp>
+#include <seqan3/range/view/take_until.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3
+{
+/*!\brief       The GenBank format.
+ * \implements  seqan3::SequenceFileInputFormat
+ * \implements  seqan3::SequenceFileOutputFormat
+ * \ingroup     sequence
+ *
+ * \details
+ *
+ * ### Introduction
+ *
+ * genbank is the format used in GenBank sequence database. See
+ * (http://quma.cdb.riken.jp/help/gbHelp.html) for a an in-depth description of the format.
+ *
+ * ### Fields
+ *
+ * The genbank format provides the fields seqan3::field::SEQ and seqan3::field::ID. Both fields are required when
+ * writing.
+ *
+ * ### Implementation notes
+ *
+ * There is no truncate_ids option while reading because the GenBank format has no (FASTA-like) idbuffer. Instead,
+ * there is the option "complete_header" to indicate whether the whole header is to be read
+ * (embl_genbank_complete_header=true) or only the "LOCUS" information should be stored.
+ *
+ * Passed qualities to either the read or write function are ignored.
```

Comment:
How can I pass qualities to the read function?

```suggestion
 * Qualities passed to the write function are ignored.
```